examples/hal_cpu_remote_server/README.md

```diff
@@ -0,0 +1,127 @@
+should consider a specific user account just for running the server that has
+limited access.
+
+We also recommend using ssh port forwarding. Note that we do not add any
+additional encryption of the executables that are sent.
+
+## Building the server
+
+This can be done as follows:
+
+```
+  cmake -GNinja <path_to_ock>/examples/hal_cpu_remote_server -Bbuild_server
+  ninja
+```
+
+If cross-compilation (for example RISC-V) is needed the following additional arguments are needed:
+
+```
+   -DCMAKE_TOOLCHAIN_FILE=<path_to_ock>/platform/riscv64-linux/riscv64-gcc-toolchain.cmake -DCMAKE_CROSSCOMPILING=TRUE
+```
+
+Note this requires the RISC-V toolchain to be installed as follows on Ubuntu:
+
+```
+  sudo apt-get install --yes gcc-9-riscv64-linux-gnu g++-9-riscv64-linux-gnu
+```
+
+If you wish to run on a non RISC-V machine, then `qemu` and the RISC-V toolchain will need to be installed.
+
+## Building the client
+
+We are looking to simplify this process, but currently we need to create an out
+of tree target. This is done as follows (for RISC-V):
+
+```
+  python <path_to_ock>/scripts/create_target.py <path_to_ock> \
+    <path_to_ock>/scripts/new_target_templates/cpu_client_riscv.json \
+    --external-dir <new_target_dir>
+
+  cd <new_target_dir>
+
+  cmake -Bbuild_client -GNinja \
+    -DCA_MUX_TARGETS_TO_ENABLE="cpu_client" \
+    -DCA_CPU_CLIENT_ENABLED=ON \
+    -DCA_ENABLE_API=cl \
+    -DCA_LLVM_INSTALL_DIR=<path_to_llvm_install> \
+    -DCA_CL_ENABLE_ICD_LOADER=ON  \
+    -DCA_EXTERNAL_ONEAPI_CON_KIT_DIR=<path_to_ock> 
+    -DCA_EXTERNAL_CPU_CLIENT_HAL_DIR=<path_to_ock>/examples/hals/hal_cpu_client
+   ninja -Cbuild_client
+```
+
+This assumes building the ICD, but it is possible to build without this.
+
+Note if we just wish a quick test then we could target `UnitCL` with the ninja
+line. Also if the server is running on a different architecture the LLVM build
+will need to enable that architecture.
+
+# Running the server
+
+The server requires a port which will be listened on. This can be any free user
+port. It also will only accept connections from specified nodes. This can be an
+ip address or host name e.g. "127.0.0.1" or "localhost". This defaults to
+"127.0.0.1", assuming that ports will be forwarded if the client is not run on
+the same machine.
+
+The server can be run as follows:
+
+```
+   cd build_server
+   ./hal_cpu_server_bin <port>
+```
+The node can also set with `-n`. It supports `-h` for info.
+
+The server will accept one connection, which when completed will end the
+program. This is to avoid programs putting the cpu into a bad state.
+
+The server can easily be run in a loop though or repeatedly started through a service manager e.g.
+
+```
+ while qemu-riscv64 ./hal_cpu_server_bin; do :; done
+```
+
+The environment variable `HAL_DEBUG_SERVER` can be set to 1 to give more debug output.
+
+We recommend this is never run as root as it runs executables sent over tcp/ip.
```

Comment:
We talked about mentioning the option of sandboxing in some way too here.

Reply:
do you have any suggestions e.g. You may also consider running in a docker to sandbox it?

Reply:
Could do that, or even just running under a user account that does not have access to anything.

Reply:
I already say that above, I've also added dockers as a mention.